libsentrykube/git.py

```diff
@@ -1,21 +1,62 @@
-def go_to_main() -> None:
-    """
-    Switches to main stashes the current work.
-    TODO: Add arguments
-    """
-    raise NotImplementedError
-
-
-def pull_main() -> None:
-    """
-    Pull main
-    TODO: Add arguments
-    """
-    raise NotImplementedError
-
-
-def create_branch() -> None:
-    """
-    TODO: Add arguments
-    """
-    raise NotImplementedError
+import git
+
+
+class RepoNotCleanException(Exception):
+    pass
+
+
+class Git:
+    def __init__(self, repo_path: str):
+        self.repo = git.Repo(repo_path)
+        if "main" in self.repo.heads:
+            self.default_branch = "main"
+        else:
+            self.default_branch = "master"
+
+    def switch_to_default_branch(self, *, force: bool) -> None:
+        if self.repo.active_branch.name == self.default_branch:
+            return
+
+        # check if there are any changes
+        if self.repo.is_dirty():
+            if not force:
+                print("Repo not clean. Cannot switch to default branch")
+                raise RepoNotCleanException
+
+            print("Stashing current work since force is true")
+            self.repo.git.stash()
+            self.stashed = True
+
+        print(f"Checking out {self.default_branch}")
+        print(self.repo.heads)
+        self.switch_to_branch(self.default_branch)
+        print(f"Active branch: {self.repo.active_branch.name}")
+        print(f"Default branch: {self.default_branch}")
+        assert self.repo.active_branch.name == self.default_branch
+
+    def stash(self) -> None:
+        self.repo.git.stash()
+        self.stashed = True
+
+    def pop_stash(self) -> None:
+        if not self.stashed:
+            return
+
+        self.repo.git.stash("pop")
+        self.stashed = False
+
+    def pull_latest_changes(self) -> None:
+        self.repo.git.pull()
+
+    def add(self, files: list[str]) -> None:
+        self.repo.index.add(files)
+
+    def commit(self, message: str) -> None:
+        self.repo.index.commit(message)
+
+    def create_branch(self, branch_name: str) -> None:
+        self.repo.create_head(branch_name, commit=self.default_branch)
+
+    def switch_to_branch(self, branch_name: str) -> None:
```

Comment:
This seems unnecessary - is there not a self.repo.git.checkout(branch_name) function?

Reply:
I could not find a function like that in the documentation. Can you point me to where you found this method?

Reply:
@nikhars https://stackoverflow.com/questions/47872070/how-to-check-out-a-branch-with-gitpython

Reply:
Thanks for sharing. Good to know. I guess there are multiple ways of achieving the same results. For now, I will keep things as is, since it seems to work.